### PR TITLE
refactor(settings): discriminate tool install actions

### LINKED ISSUE #11664: Replace optional tool installation fields with a tagged install action
## Context

A scheduled data-structure complexity audit found that `ToolDashboardItemSchema` in `src/shared/schemas/settingsViewMessages.ts` exposes independent optional `installGuide`, `installUrl`, `installExtensionId`, `installCommand`, and `authCommand` fields. `ToolCard.ts` must probe those fields in precedence-ordered `if` chains because the type does not state which combinations are valid.

## Scope

- Introduce one tagged installation/action union, for example URL, extension, command, authentication command, or guide-only variants.
- Make the producer select one explicit variant.
- Make `ToolCard` render by the variant discriminant rather than optional-field precedence.
- Preserve current user-visible actions and copy.

## Acceptance

- Invalid installation field combinations are unrepresentable or rejected at the message schema boundary.
- Existing tool cards retain their current primary actions and fallback guidance.
- Focused settings-view tests, typechecks, lint, architecture, and dead-code ratchets pass.

### PR DESCRIPTION
**Pull request by LionSR**
## Summary

Replace the settings-view tool card's five independent optional installation fields with an `installActions` array whose entries use one discriminated union. The dashboard producer now emits explicit guide, URL, extension, install-command, and authentication-command variants, and the card renders and routes each variant by `kind` while preserving the existing copy, action order, primary styling, and desktop extension filtering.

Closes #11664.

## Issue acceptance gates

- Invalid per-action field combinations are rejected by strict discriminated-union branches at the settings-view outbound schema boundary.
- Legacy independent installation fields are rejected instead of becoming a second representation.
- Existing guide, URL, extension, terminal-install, and terminal-auth actions retain their rendering and routes.
- Focused settings-view, desktop, schema-boundary, architecture, typecheck, lint, dead-code, and compile checks pass.

## Single source of truth

`ToolInstallActionSchema` in `src/shared/schemas/settingsViewMessages.ts` is the canonical settings-view wire representation for tool setup actions.

## Duplication and abstraction budget

Removed five optional fields and the corresponding precedence/probing logic from `ToolCard`. The new union owns the invariant that each action kind carries exactly its required payload and no fields from another kind. No pass-through layer was added.

## Net elements (R6)

- Files: 1 added, 6 modified
- Diffstat: +277 / -114, net +163 lines (including focused regression coverage)
- Exported symbols: +1 type (`ToolInstallAction`), -0
- Classes/interfaces/enums: no additions or deletions

## Consumer counts (R8)

n/a. No emit path, event key, or export was deleted or rerouted. The new exported type has one production consumer (`ToolCard`); the schema remains colocated with the existing `ToolDashboardItem` contract.

## Host impact

Extension: Tool cards render and dispatch setup actions from the discriminated action list.

Desktop: Filters the `extension` action variant while retaining every standalone setup action.

CLI: None. CLI tool definitions and doctor output keep their existing host-specific representation.

## Scheduled refactoring

None.

## Validation

Passed:

- `corepack pnpm vitest run --config vitest.config.mjs src/test-kernel/settings/ToolCard.vitest.ts src/test-kernel/settings/ToolsTab.vitest.ts src/test-kernel/settings/PlanToolTerminalAction.vitest.ts src/test-kernel/desktop/DesktopToolingSettingsController.vitest.ts` (4 files, 22 tests)
- `corepack pnpm vitest run --config vitest.config.mjs src/test-kernel/architecture` (15 files, 101 tests)
- `npm run typecheck` (all six typecheck lanes)
- `npm run lint`
- changed-file Prettier check
- changed-file ESLint check
- `npm run check:dead-code-ratchet`
- `npm run compile:fast`
- `git diff --check`

The full `npm test` run completed 820 files / 9,913 tests successfully but was not green: 3 files failed (4 tests), plus 2 suite hook timeouts. The two remaining assertion failures in `src/test-kernel/desktop/DesktopAgentExecution.vitest.ts` reproduce on untouched `main` at `6da093094f`; the hook timeouts are load-sensitive and the affected suites pass when rerun outside the full-suite contention. None of those files are changed here.

### CODE REVIEW

File: src/test-kernel/settings/ToolCard.vitest.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+};
+
+function dashboardMessage(installActions: unknown) {
+  return {
+    command: SETTINGS_VIEW_COMMANDS.UPDATE_TOOL_DASHBOARD,
+    items: [{ ...ITEM, installActions }],
+  };
+}
+
+describe('tool-card install actions', () => {
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/test-kernel/settings/ToolCard.vitest.ts]: Fixed in 3120f38ef1. Deleted the new ToolCard component suite and its redundant rendering/routing coverage. The issue-required malformed tagged-action boundary cases now live as one table-driven test in the existing shared schema suite.

File: packages/extension/src/settingsView/frontend/components/tools/ToolCard.ts [hunk 2]
```diff
@@ -262,21 +250,86 @@ export class ToolCard extends LitElement {
...
+  }
 
-    const hasGuide =
-      this.item.installGuide ||
-      this.item.installUrl ||
-      this.item.installCommand ||
-      this.item.authCommand;
-    if (!hasGuide) return nothing;
+  private renderGuide(): TemplateResult | typeof nothing {
+    if (!this.item.requiresSetup || this.item.installActions.length === 0) {
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/extension/src/settingsView/frontend/components/tools/ToolCard.ts]: Fixed in 3120f38ef1. The setup-details gate now ignores an extension-only action list, matching the old installExtensionId-excluded gate, while Lean still renders because it also carries guide and URL actions.

**Comment by LionSR** [comment]
@codex review
